### ISSUE TITLE
Terraform crash -- panic: inconsistent list element types

### ISSUE THREAD

**Issue by vijayrajah** [hashicorp/terraform#26195]
### Terraform Version
```
Terraform version: 0.13.2
```

### Crash Output
2020/09/10 05:08:31 [INFO] Terraform version: 0.13.2
2020/09/10 05:08:31 [INFO] Go runtime version: go1.14.7
2020/09/10 05:08:31 [INFO] CLI args: []string{"/data/home/myuser/local/bin/tf-0.13.2", "plan"}
2020/09/10 05:08:31 [DEBUG] Attempting to open CLI config file: /home/myuser/.terraformrc
2020/09/10 05:08:31 [DEBUG] File doesn't exist, but doesn't need to. Ignoring.
2020/09/10 05:08:31 [DEBUG] ignoring non-existing provider search directory terraform.d/plugins
2020/09/10 05:08:31 [DEBUG] ignoring non-existing provider search directory /home/myuser/.terraform.d/plugins
2020/09/10 05:08:31 [DEBUG] ignoring non-existing provider search directory /home/myuser/.local/share/terraform/plugins
2020/09/10 05:08:31 [DEBUG] ignoring non-existing provider search directory /usr/local/share/terraform/plugins
2020/09/10 05:08:31 [DEBUG] ignoring non-existing provider search directory /usr/share/terraform/plugins
2020/09/10 05:08:31 [INFO] CLI command args: []string{"plan"}
panic: inconsistent list element types (cty.Object(map[string]cty.Type{"access":cty.String, "description":cty.String, "destination_address_prefix":cty.String, "destination_application_security_group_ids":cty.DynamicPseudoType, "destination_port_range":cty.String, "destination_port_ranges":cty.DynamicPseudoType, "direction":cty.String, "name":cty.String, "priority":cty.String, "protocol":cty.String, "source_address_prefix":cty.String, "source_application_security_group_ids":cty.DynamicPseudoType, "source_port_range":cty.String, "source_port_ranges":cty.DynamicPseudoType}) then cty.Object(map[string]cty.Type{"access":cty.String, "description":cty.String, "destination_address_prefix":cty.String, "destination_application_security_group_ids":cty.DynamicPseudoType, "destination_port_range":cty.String, "destination_port_ranges":cty.Tuple([]cty.Type{cty.String, cty.String}), "direction":cty.String, "name":cty.String, "priority":cty.String, "protocol":cty.String, "source_address_prefix":cty.String, "source_application_security_group_ids":cty.DynamicPseudoType, "source_port_range":cty.String, "source_port_ranges":cty.DynamicPseudoType}))

goroutine 1 [running]:
github.com/zclconf/go-cty/cty.ListVal(0xc00027a460, 0x5, 0x5, 0xc0000cd800, 0x3, 0x4, 0x1)
        /home/circleci/project/project/vendor/github.com/zclconf/go-cty/cty/value_init.go:166 +0x436
github.com/zclconf/go-cty/cty/convert.conversionTupleToList.func2(0x2cf0500, 0xc00061c240, 0x21d18a0, 0xc00061c260, 0xc00061c4a0, 0x2, 0x2, 0x13e000002492101, 0xc0004be910, 0x413f63, ...)
        /home/circleci/project/project/vendor/github.com/zclconf/go-cty/cty/convert/conversion_collection.go:319 +0x7df
github.com/zclconf/go-cty/cty/convert.getConversion.func1(0x2cf0500, 0xc00061c240, 0x21d18a0, 0xc00061c260, 0xc00061c4a0, 0x2, 0x2, 0x21d18a0, 0xc00061c260, 0x0, ...)
        /home/circleci/project/project/vendor/github.com/zclconf/go-cty/cty/convert/conversion.go:46 +0x42b
github.com/zclconf/go-cty/cty/convert.conversionObjectToObject.func1(0x2cf0480, 0xc00000e188, 0x2306c40, 0xc0001bd260, 0xc00061c4a0, 0x2, 0x2, 0x40c601, 0x403d44, 0xc0000c6390, ...)
        /home/circleci/project/project/vendor/github.com/zclconf/go-cty/cty/convert/conversion_object.go:65 +0x450
github.com/zclconf/go-cty/cty/convert.getConversion.func1(0x2cf0480, 0xc00000e188, 0x2306c40, 0xc0001bd260, 0xc00048ce20, 0x1, 0x1, 0x2306c40, 0xc0001bd260, 0x23be060, ...)
        /home/circleci/project/project/vendor/github.com/zclconf/go-cty/cty/convert/conversion.go:46 +0x42b
github.com/zclconf/go-cty/cty/convert.conversionObjectToMap.func2(0x2cf0480, 0xc00000e190, 0x2306c40, 0xc0001bd770, 0x0, 0x0, 0x0, 0xc0003b8700, 0x10, 0x10, ...)
        /home/circleci/project/project/vendor/github.com/zclconf/go-cty/cty/convert/conversion_collection.go:382 +0x534
github.com/zclconf/go-cty/cty/convert.getConversion.func1(0x2cf0480, 0xc00000e190, 0x2306c40, 0xc0001bd770, 0x0, 0x0, 0x0, 0xc00059c5d0, 0xc00048ce10, 0x2cf0440, ...)
        /home/circleci/project/project/vendor/github.com/zclconf/go-cty/cty/convert/conversion.go:46 +0x42b
github.com/zclconf/go-cty/cty/convert.retConversion.func1(0x2cf0480, 0xc00000e190, 0x2306c40, 0xc0001bd770, 0xc00048ce10, 0x0, 0x0, 0x0, 0x0, 0x0)
        /home/circleci/project/project/vendor/github.com/zclconf/go-cty/cty/convert/conversion.go:188 +0x6b
github.com/zclconf/go-cty/cty/convert.Convert(0x2cf0480, 0xc00000e190, 0x2306c40, 0xc0001bd770, 0x2cf0440, 0xc00007d340, 0x0, 0x0, 0x0, 0xc00007d310, ...)
        /home/circleci/project/project/vendor/github.com/zclconf/go-cty/cty/convert/public.go:51 +0x1aa
github.com/hashicorp/terraform/configs.decodeVariableBlock(0xc000521790, 0x0, 0x0, 0x0, 0x0, 0x0)
        /home/circleci/project/project/configs/named_values.go:109 +0x8c1
github.com/hashicorp/terraform/configs.(*Parser).loadConfigFile(0xc000481f80, 0xc0000c68e0, 0xc, 0x0, 0x2, 0xc0001c50c0, 0x1, 0x2)
        /home/circleci/project/project/configs/parser_config.go:112 +0xdc2
github.com/hashicorp/terraform/configs.(*Parser).LoadConfigFile(...)
        /home/circleci/project/project/configs/parser_config.go:22
github.com/hashicorp/terraform/configs.(*Parser).loadFiles(0xc000481f80, 0xc000564700, 0x3, 0x4, 0x0, 0x4, 0x0, 0x0, 0x0, 0x0, ...)
        /home/circleci/project/project/configs/parser_config_dir.go:77 +0x342
github.com/hashicorp/terraform/configs.(*Parser).LoadConfigDir(0xc000481f80, 0x26cd2e5, 0x1, 0x0, 0x1, 0x0, 0xc00063f938)
        /home/circleci/project/project/configs/parser_config_dir.go:37 +0x121
github.com/hashicorp/terraform/command.(*Meta).loadSingleModule(0xc000312700, 0x26cd2e5, 0x1, 0xc00063f938, 0x20208cc, 0xc000520750, 0x1)
        /home/circleci/project/project/command/meta_config.go:113 +0x135
github.com/hashicorp/terraform/command.(*Meta).loadBackendConfig(0xc000312700, 0xc000052054, 0x43, 0x0, 0x0, 0x0, 0x0)
        /home/circleci/project/project/command/meta_config.go:175 +0x43
github.com/hashicorp/terraform/command.(*PlanCommand).Run(0xc000312700, 0xc0000bc030, 0x0, 0x0, 0xc0000a7870)
        /home/circleci/project/project/command/plan.go:69 +0x7f8
github.com/mitchellh/cli.(*CLI).Run(0xc000562280, 0xc000562280, 0xc00063fd48, 0x1)
        /home/circleci/project/project/vendor/github.com/mitchellh/cli/cli.go:255 +0x1da
main.wrappedMain(0x0)
        /home/circleci/project/project/main.go:275 +0xc90
main.realMain(0x0)
        /home/circleci/project/project/main.go:106 +0xde
main.main()
        /home/circleci/project/project/main.go:42 +0x3a

### Expected Behavior

The variable here is a Map with 2 elements. Each of the 2 elements have multiple K, V pairs

This is the full Variable definition is below. This should be parsed. This map is passed to 'azurerm_network_security_group' resource

The crash happens with 0.13.2, 0.13.1, 0.13.0 & 0.12.24 (versions of TF-CLI)
```
variable "subnet_details" {
  type        = map
  description = "Subnet Details"
  default = {
    "subnet1" = {
      "name"                    = "subnet1"
      "address_space"           = "172.16.0.0/16"
      "nsg_name"                = "subnet1-nsg"
      "service_delegation_name" = null
      "service_actions"         = null
      "subnet_nsg_rules" = [
        {
          name                                       = "allow_outbound_internet_80"
          description                                = "Allow internet access to HTTP "
          priority                                   = "100"
          direction                                  = "Outbound"
          access                                     = "Allow"
          protocol                                   = "Tcp"
          source_port_range                          = "*"
          source_port_ranges                         = null
          source_address_prefix                      = "VirtualNetwork"
          source_application_security_group_ids      = null
          destination_address_prefix                 = "Internet"
          destination_port_range                     = 80
          destination_port_ranges                    = null
          destination_application_security_group_ids = null
        },
        {
          name                                       = "allow_outbound_internet_443"
          description                                = "Allow internet access to HTTPS "
          priority                                   = "101"
          direction                                  = "Outbound"
          access                                     = "Allow"
          protocol                                   = "Tcp"
          source_port_range                          = "*"
          source_port_ranges                         = null
          source_address_prefix                      = "VirtualNetwork"
          source_application_security_group_ids      = null
          destination_address_prefix                 = "Internet"
          destination_port_range                     = 443
          destination_port_ranges                    = null
          destination_application_security_group_ids = null
        },
        {
          name                                       = "allow_outbound_internet_123"
          description                                = "Allow internet access to NTP "
          priority                                   = "102"
          direction                                  = "Outbound"
          access                                     = "Allow"
          protocol                                   = "Udp"
          source_port_range                          = "*"
          source_port_ranges                         = null
          source_address_prefix                      = "VirtualNetwork"
          source_application_security_group_ids      = null
          destination_address_prefix                 = "Internet"
          destination_port_range                     = 123
          destination_port_ranges                    = null
          destination_application_security_group_ids = null
        },
        {
          name                                       = "allow_windows_activcation"
          description                                = "allow_windows_activcation "
          priority                                   = "103"
          direction                                  = "Outbound"
          access                                     = "Allow"
          protocol                                   = "Tcp"
          source_port_range                          = "*"
          source_port_ranges                         = null
          source_address_prefix                      = "VirtualNetwork"
          source_application_security_group_ids      = null
          destination_address_prefix                 = "23.102.135.246/32"
          destination_port_range                     = "*"
          destination_port_ranges                    = null
          destination_application_security_group_ids = null
        },
        {
          name                                       = "allow_postgres_access"
          description                                = "allow_postgres_access "
          priority                                   = "104"
          direction                                  = "Outbound"
          access                                     = "Allow"
          protocol                                   = "Tcp"
          source_port_range                          = "*"
          source_port_ranges                         = null
          source_address_prefix                      = "VirtualNetwork"
          source_application_security_group_ids      = null
          destination_address_prefix                 = "sql.eastus2"
          destination_port_range                     = "5432"
          destination_port_ranges                    = null
          destination_application_security_group_ids = null
        },
        {
          name                                       = "allow_internal_vnet_access"
          description                                = "allow_intervnet_access"
          priority                                   = "105"
          direction                                  = "Outbound"
          access                                     = "Allow"
          protocol                                   = "Tcp"
          source_port_range                          = "*"
          source_port_ranges                         = null
          source_address_prefix                      = "172.16.0.0/16"
          source_application_security_group_ids      = null
          destination_address_prefix                 = "172.16.0.0/16"
          destination_port_range                     = null
          destination_port_ranges                    = ["10000","18092","8092","9083"]
          destination_application_security_group_ids = null
        },
        {
          name                                       = "allow_databricks_vnet_access"
          description                                = "allow_databricks_access"
          priority                                   = "901"
          direction                                  = "Outbound"
          access                                     = "Allow"
          protocol                                   = "Tcp"
          source_port_range                          = "*"
          source_port_ranges                         = null
          source_address_prefix                      = "172.16.0.0/16"
          source_application_security_group_ids      = null
          destination_address_prefix                 = "172.20.0.0/16"
          destination_port_range                     = "*"
          destination_port_ranges                    = null
          destination_application_security_group_ids = null
        },
        {
          name                                       = "disable_vnet_access"
          description                                = "Disable Vnet Acess"
          priority                                   = "4000"
          direction                                  = "Outbound"
          access                                     = "Deny"
          protocol                                   = "*"
          source_port_range                          = "*"
          source_port_ranges                         = null
          source_address_prefix                      = "VirtualNetwork"
          source_application_security_group_ids      = null
          destination_address_prefix                 = "VirtualNetwork"
          destination_port_range                     = "*"
          destination_port_ranges                    = null
          destination_application_security_group_ids = null
        },
        {
          name                                       = "disable_internet_access"
          description                                = "Disable internet Acess"
          priority                                   = "4001"
          direction                                  = "Outbound"
          access                                     = "Deny"
          protocol                                   = "*"
          source_port_range                          = "*"
          source_port_ranges                         = null
          source_address_prefix                      = "VirtualNetwork"
          source_application_security_group_ids      = null
          destination_address_prefix                 = "Internet"
          destination_port_range                     = "*"
          destination_port_ranges                    = null
          destination_application_security_group_ids = null
        }
      ]
    },
    "appgwy-sn" = {
      "name"                    = "appgwy-sn"
      "address_space"           = "172.32.255.0/24"
      "nsg_name"                = "appgwy-sn-nsg"
      "service_delegation_name" = null
      "service_actions"         = null
      "subnet_nsg_rules" = [
        {
          name                                       = "ALLOW-GATEWAYMANAGER-IN"
          description                                = "ALLOW-GATEWAYMANAGER-IN"
          priority                                   = "100"
          direction                                  = "Inbound"
          access                                     = "Allow"
          protocol                                   = "Tcp"
          source_port_range                          = "*"
          source_port_ranges                         = null
          source_address_prefix                      = "GatewayManager"
          source_application_security_group_ids      = null
          destination_address_prefix                 = "*"
          destination_port_range                     = "65200-65535"
          destination_port_ranges                    = null
          destination_application_security_group_ids = null
        },
        {
          name                                       = "allow_http_80_443_in"
          description                                = "allow_http_80_443_in"
          priority                                   = "101"
          direction                                  = "Inbound"
          access                                     = "Allow"
          protocol                                   = "Tcp"
          source_port_range                          = "*"
          source_port_ranges                         = null
          source_address_prefix                      = "AzureFrontDoor.Backend"
          source_application_security_group_ids      = null
          destination_address_prefix                 = "172.32.255.0/24" #entire subnet
          destination_port_range                     = null
          destination_port_ranges                    = 80
          destination_application_security_group_ids = null
        },
        {
          name                                       = "allow_https_443_in"
          description                                = "allow_https_443_in"
          priority                                   = "102"
          direction                                  = "Inbound"
          access                                     = "Allow"
          protocol                                   = "Tcp"
          source_port_range                          = "*"
          source_port_ranges                         = null
          source_address_prefix                      = "AzureFrontDoor.Backend"
          source_application_security_group_ids      = null
          destination_address_prefix                 = "172.32.255.0/24" #entire subnet
          destination_port_range                     = 443
          destination_port_ranges                    = null
          destination_application_security_group_ids = null
        },
        {  ###### NOW OUTBOUND RULES ####################
          name                                       = "allow_http_80_out"
          description                                = "allow_http_80_out "
          priority                                   = "100"
          direction                                  = "Outbound"
          access                                     = "Allow"
          protocol                                   = "Tcp"
          source_port_range                          = "*"
          source_port_ranges                         = null
          source_address_prefix                      = "VirtualNetwork"
          source_application_security_group_ids      = null
          destination_address_prefix                 = "VirtualNetwork"
          destination_port_range                     = "80"
          destination_port_ranges                    = null
          destination_application_security_group_ids = null
        },
        {  
          name                                       = "allow_http_443_out"
          description                                = "allow_http_443_out "
          priority                                   = "101"
          direction                                  = "Outbound"
          access                                     = "Allow"
          protocol                                   = "Tcp"
          source_port_range                          = "*"
          source_port_ranges                         = null
          source_address_prefix                      = "VirtualNetwork"
          source_application_security_group_ids      = null
          destination_address_prefix                 = "VirtualNetwork"
          destination_port_range                     = "443"
          destination_port_ranges                    = null
          destination_application_security_group_ids = null
        }
      ]
    }
  }
}```

**Comment by pselle** [comment]
Hello! The error message in the panic contains the information you need to work around the bug, but Terraform should display an error message and not a panic. I've reproduced using your [config](https://github.com/danieldreier/terraform-issue-reproductions/tree/master/26195) and will label this as a bug.

As to working around the bug, you must use consistent types in a collection as noted here: https://www.terraform.io/docs/configuration/types.html#collection-types. This is why you're seeing the error.

**Comment by vijayrajah** [comment]
@pselle Thanks, Yes, I have fixed this error in my var deceleration.

**Comment by mildwonkey** [comment]
The linked go-cty PR, which closes this issue, is now included in terraform's main branch (and 0.15) 🎉 
This same configuration now results in an error instead of a panic:

```
│ This default value is not compatible with the variable's type constraint: element types must all match
│ for conversion to list.
```

I'm going to close this, thanks again for reporting it!
